### PR TITLE
Reverse study mode with locations on offer summary cards

### PR DESCRIPTION
**Pull request by despo**
## Context
Update components displaying requested of offered course details in the order displayed in the make offer flow.

## Changes proposed in this pull request

Update offer summary, course summary and old make offer components to first display the study mode and then the location of a course.

## Guidance to review

Changes should be reflected on
1. Make offer decision page
2. Offer summary page (at the end of make or change offer flow)
3. Offer tab
4. Defer & withdraw offer (can be accessed via offer tab when application choice is in correct state)
4. Reconfirm offer (end of defer offer flow)

## Link to Trello card

https://trello.com/c/F9HCtsg2/3618-reverse-study-mode-with-locations-on-offer-summary-cards

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/provider_interface/completed_offer_summary_component.rb [hunk 1]
```diff
@@ -12,14 +12,14 @@ def rows
...
           change_path: change_course_path },
-        { key: 'Full time or part time',
-          value: course_option.study_mode.humanize,
-          action: 'Change',
-          change_path: change_study_mode_path },
         { key: 'Location',
           value: course_option.site.name_and_address,
           action: 'Change',
           change_path: change_location_path },
+        { key: 'Full time or part time',
```
Comment (mnacos) [inline app/components/provider_interface/completed_offer_summary_component.rb:19]: I'm a little confused, the card says 'Full time/part time should come before locations in all summary displays'. Is this change required?

File: app/components/provider_interface/read_only_completed_offer_summary_component.rb [hunk 2]
```diff
@@ -8,10 +8,10 @@ def rows
           value: course_option.provider.name_and_code },
         { key: 'Course',
           value: course_option.course.name_and_code },
-        { key: 'Full time or part time',
-          value: course_option.study_mode.humanize },
         { key: 'Location',
           value: course_option.site.name_and_address },
+        { key: 'Full time or part time',
```
Comment (mnacos) [inline app/components/provider_interface/read_only_completed_offer_summary_component.rb:13]: Same, the previous order seems correct.

File: spec/components/provider_interface/completed_offer_summary_component_spec.rb [hunk 3]
```diff
@@ -21,8 +21,8 @@ def row_text_selector(row_name, render)
     rows = {
       provider: 0,
       course: 1,
-      full_or_part_time: 2,
-      location: 3,
+      location: 2,
+      full_or_part_time: 3,
```
Comment (mnacos) [inline spec/components/provider_interface/completed_offer_summary_component_spec.rb:25]: Same question

**Review by mnacos — COMMENTED** [review]
Code-wise it's fine, but could we double-check the order is correct in all cases? If the order in all components is correct, maybe we need to update Trello.

File: app/components/provider_interface/completed_offer_summary_component.rb (hunk 1 above)
Comment (despo) [reply app/components/provider_interface/completed_offer_summary_component.rb]: I think in my attempt to get through all the make offer components I missed that some had been implemented with the details in the correct order :)

Thanks for flagging. Sorted now.

**Review by cwrw — APPROVED** [review]
⏬  looks good, only question is do we want to also do this while we're at it? https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/components/provider_interface/deferred_offer_details_component.rb#L28
